quickstarts/quickstart-template-guide.md

```diff
@@ -0,0 +1,141 @@
+
+## Introduction
+
+A quickstart introduces your users to your application for the first time.
+It focuses on the **primary feature** of the application and helps your users to start using the application as quickly as possible.
+
+A good quickstart answers the following questions:
+
+- Scope:
+ - What is the core purpose of this application?
+ - What is the minimal use case that is covered in this quickstart?
+- Install (if applicable):
+ - How do I download, install, and configure the application?
+ - How do I install and configure the application?
+ - How can I get any required access keys or authentication credentials?
+- Hello  World:
+ - How can I run a simple workflow for a core feature or a common use? case.
+- Next steps:
+ - What other features are available to explore in the application?
+
+A Quickstart guide is often confused with a Getting Started  guide. Though both these documents help the users get acquainted with the application they widely differ in their target audience. 
+
+There are instances where a Marketing guide is often confused with a Quickstart and a Getting Started guide. The inline table gives a brief comparison about the differences. 
+
+| |Quickstart Guide|Getting Started Guide| Marketing Guide|
+|--------|----------------|----------------------|---------------|
+ |Target Audience|Domain Experts: Know the problem space and are aware of what they need to do|Beginners:New to problem space and the product|Business Decision Maker: Makes strategic decisions on whether to buy the product 
+|Content|Minimal conceptual information|Detailed conceptual information on product/domain| No/less conceptual information. Instead focuses on the benefits and the customers using the product.
+|Focus|how-to| how-to and why|  sales
+
+## Why do I need a quickstart?
+
+A quickstart is often the first opportunity for your user to build an opinion about the technical quality of your project. It can:
+- Make users comfortable using the application.
+- Work with any application version the user might use.
+- Lead to the willingness to handle complex workflows.
+- Help the users in exploring additional features.
+- Improve the customer experience, and help reduce costs by lowering the number of support requests.
+
+## Tips for writing a quickstart
+
+- Highlight the primary feature of your application.
+- Document the quickest/easiest way to implement the primary feature  of your application.
+- Pick a use case that the user can complete within 1-2 hours with a preference for a shorter time.
+-  Avoid complicating the quickstart by including error scenarios/complex use cases.
+- Lengthy quickstarts can overwhelm users. Consider condensing or removing steps or reevaluating the scope of the quickstart.
+- For code samples, ensure that you include:  
+ - Any required `import` or `using` statements.
+ - Code comments that explain what the code does.
```

Comment:
**Suggestion (blocking):** It seems to me like this bulleted list is too much of a set of miscellaneous best practices.

I'd suggest simplifying the list by breaking it up by purpose. For example, I would add a new section that could talk about pre-writing steps such as how to pick the right task or activity that the user will complete in the quickstart. (You could also build out a larger version of this section in your deep dive document.) Some of your bullet points talk about how to select this task:

- Highlight the primary feature of your application.
- Document the quickest/easiest way to implement the primary feature  of your application.
- Pick a use case that the user can complete within 1-2 hours with a preference for a shorter time.

Whereas some of the others are just general best practices (and I recommend renaming it to best practices rather than tips):
-  Avoid complicating the quickstart by including error scenarios/complex use cases.
- Lengthy quickstarts can overwhelm users. Consider condensing or removing steps or reevaluating the scope of the quickstart.
- For code samples, ensure that you include:  
 - Any required `import` or `using` statements.
 - Code comments that explain what the code does.
 
Formatting issue (blocking): When I look at the rendered version of this markdown, the bullets don't have the right hierarchy. They all appear to be on the same level as opposed to some of the bullets being listed underneath the higher level bullets. The problem is that it's hard to see the dependent relationships in the content here, which makes it confusing.

Likely this is due to not having enough spaces in front of the nested bullet points. Fix the spacing and then try looking at the rendered bulleted list (either in your code editor or on a free Markdown previewer online) and make sure that the bullet levels are correct.

Reply:
Fixed